snuba/datasets/entities/sessions.py

```diff
@@ -250,3 +257,59 @@ def get_query_processors(self) -> Sequence[QueryProcessor]:
                 {"bucketed_started": "started"}, ("started", "received")
             ),
         ]
+
+
+GRANULARITY_MAPPING = {
+    "toStartOfMinute": 60,
+    "toStartOfHour": 3600,
+    "toDate": 86400,
+}
+
+
+def extract_granularity_from_query(query: Query) -> Optional[int]:
+    """
+    This extracts the `granularity` from the `groupby` statement of the query,
+    if the query is grouped by the `bucketed_started` virtual column
+    (which is a transformation of the `started` column).
```

Comment:
I am still not sure how you would like to expose the `started` column to the user. Today the client does not use it (and we could hide that if needed), though if the single entity wraps two storages, the behavior of the `started` column exposed is inconsistent. It would be already bucketed when looking at the pre-aggregated table and it would be the raw started timestamp for the raw table. 
I think we should either hide the column to the client or expecting it to be treated as a raw timestamp, which means expecting it to be an argument of toStartOfHour (and similar functions) if the client wants to refer to the aggregated data. 
Not a big issue as the column is not used by the client today.

Reply:
I would just hide it, as I don’t think it makes sense to query for that directly.